src/components/AdministrationPage/AdministrationPage.tsx

```diff
@@ -158,32 +193,107 @@ export const AdministrationPage = () => {
+                        fieldId="form-announcement"
+                        helperTextInvalid={invalidAnnouncement() ? 'Required field.' : null}
+                        helperTextInvalidIcon={<ExclamationCircleIcon />}
+                        validated={'error'}
+                      >
                         <TextArea
                           name="form-announcement"
                           id="form-announcement"
                           value={announcementMessage}
                           onChange={(value: string) => {
                             setAnnouncementMessage(value);
                           }}
+                          onBlur={() => {
+                            setAnnouncementTouched(true);
+                          }}
                         />
                       </FormGroup>
                     </GridItem>
+                    {isMaintenanceModeOn && (
+                      <GridItem span={12}>
+                        <FormGroup
+                          label="* ETA Time"
+                          fieldId="eta-time"
+                          helperTextInvalid={invalidEtaTime() ? 'Required field.' : null}
+                          helperTextInvalidIcon={<ExclamationCircleIcon />}
+                          validated={'error'}
+                        >
+                          <DatePicker
+                            required
+                            isDisabled={!isMaintenanceModeOn || isEtaNa}
+                            name="form-etaTime"
+                            id="eta-time"
+                            placeholder="yyyy-MM-dd hh:mm (UTC)"
+                            dateFormat={transDateFormat}
+                            onClick={() => {
+                              setEtaTouched(true);
+                            }}
+                            dateParse={(dateString) => {
+                              return new Date(dateString);
+                            }}
+                            value={etaTime}
+                            onBlur={(value: string) => {
+                              setEtaTime(value);
+                            }}
+                            onChange={(value: string) => {
+                              setEtaTime(value);
+                            }}
+                            aria-invalid={invalidEtaTime()}
+                          />
+                          &nbsp;&nbsp;
+                          <Switch
+                            id="eta-na-switch"
+                            label=" N/A:"
+                            labelOff=" N/A:"
+                            hasCheckIcon
+                            isChecked={isEtaNa}
+                            onChange={() => {
+                              setIsEtaNa(!isEtaNa);
+                              setEtaTouched(true);
+                            }}
+                            isReversed
+                          />
```

Comment:
Bug:
If date is already set and you switch N/A state to true, announcement will still include date and not 'N/A'.